backend/rust/models/src/lib.rs

```diff
@@ -0,0 +1,13 @@
+pub(crate) mod mnist;
+
+use bunker::pb::{
+    ModelOptions,
+    PredictOptions
+};
+
+pub trait LLM {
```

Comment:
Hi @lu-zero, I have a question here. I believe we can define a trait for example "LLM" for "llama", "whisper" to implement. So, we can call a generic type in the main function. Is this a good idea?

Are there a better ways to implement this idea? I want to use a generic type in the main function, and there no need to edit code as more models are implemented. Here you can see, I have been defined a trait but I do not know how to deal with LLM in `backend-burn`'s predict function. Thanks.

Reply:
How do you want to instantiate it? 
You can use `Box<dyn Trait>` and use the dynamic dispatch if it is negligible or you can use the enum-dispatch in case (see https://docs.rs/enum_dispatch/latest/enum_dispatch/ for an easy way to implement that pattern)

Reply:
Thank you, this is cool. Here is my thought.

I assume that we received an gRPC request in `async fn predict(&self, request: Request<PredictOptions>) -> Result<Response<Reply>, Status>` function. Here is an example:

```rust
async fn predict(&self, request: Request<PredictOptions>) -> Result<Response<Reply>, Status>{
   # LLM can be `Llama2`, `whisper`, `mnist`
   let instance=LLM.new()
   let result=instance.predict(request.into_inner().promt)
  Ok(result)
}
```